actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -161,36 +144,12 @@ def javascript_include_tag(*sources)
       #   # => <link href="/assets/random.styles" rel="stylesheet" />
       #   #    <link href="/css/stylish.css" rel="stylesheet" />
       def stylesheet_link_tag(*sources)
-        options = sources.extract_options!.stringify_keys
-        path_options = options.extract!("protocol", "extname", "host", "skip_pipeline").symbolize_keys
-        preload_links = []
-        options["crossorigin"] = "anonymous" if options["crossorigin"] == true
-        nopush = options["nopush"].nil? ? true : options.delete("nopush")
-
-        sources_tags = sources.uniq.map { |source|
-          href = path_to_stylesheet(source, path_options)
-          if preload_links_header && href.present? && !href.start_with?("data:")
-            preload_options = { rel: "preload", as: "style"}
-            params = tag.attributes(preload_options.merge!(options)).split(" ").join("; ")
-            preload_links << "<#{href}>; #{params}#{"; nopush" if nopush}"
-          end
-          tag_options = {
-            "rel" => "stylesheet",
-            "href" => href
-          }.merge!(options)
-
+        multiple_assets_tag_builder("stylesheet", sources, tag_name: :link, preload_as: :style) do |tag_options|
+          tag_options["rel"] = "stylesheet"
```

Comment:
This pattern feels weird to me, vs something like `sources.merge(rel: "stylesheet")` on the line above. Is there something I'm missing about why that approach wouldn't work?